contents/testing/security-testing.md

```diff
@@ -0,0 +1,93 @@
+
+The goal of security testing is to:
+- Identify threats in the system.
+- To measure the potential vulnerabilities of the system.
+- To help in detecting every possible security risks in the system.
+- To help developers in fixing the security problems through coding.
+
+### Types of Security Testing
+- **Vulnerability Scanning**: This is done through automated software to scan a system against known vulnerability signatures.
+- **Security Scanning**: It involves identifying network and system weaknesses, and later provides solutions for reducing these risks. This scanning can be performed both manually and automatically.
+- **Penetration Testing**: This kind of testing simulates an attack from a malicious hacker. This testing involves analysis of a particular system to check for potential vulnerabilities to an external hacking attempt.
+- **Risk Assessment**: This testing involves analysis of security risks observed in the organization. Risks are classified as  Low, Medium and High. This testing recommends controls and measures to reduce the risk.
+- **Security Auditing**: This is an internal inspection of Applications and Operating systems for security flaws. An audit can also be done via line by line inspection of code.
+- **Posture Assessment**: This combines Security Scanning, Penetration Testing, and Risk Assessments to show an overall security posture of an organization.
+
+## Why Have Security Testing?
+
+A comprehensive security testing framework deals with validation across all layers of an application. Through this, the organization can evaluate their application code for vulnerabilities and take remedial measures for the same. Recently, many of the software development organizations have been making use of secure software development life cycle methodologies to ensure identification and rectification of vulnerability areas early on in the application development process.
+
+Security Testing helps to ensure:
+- Confidentiality of Information.
+- Integrity of Information.
+- Authenticity of Identity.
+- Availability of information and services.
+- Authorisation: The process of determining whether a requester is allowed to receive a service or perform an operation (e.g. Access Control).
+
+## How to Get Started with Security Testing?
+
+As a subset of software testing, security testing operates in tandem with your SDLC. It is a conscious decision that needs to be made for a more secure application.
+
+![Generic SDLC Model](security-testing/Sec-Test-Fig-1.png "Generic SDLC Model")
+
+<box id="info">:fas-info-circle: Knowledge of Software Development Life-Cycles (SDLCs) would be useful but not necessary. If you would like to find out more about SDLC Processes please click [here](https://nus-cs2103-ay1920s1.github.io/website/se-book-adapted/chapters/processModels.html).</box>
```

Comment:
It is possible to omit SDLC from the article altogether. Just say that security testing needs to be done at different phases of a software project.

Reply:
I am in the midst of re-writing this article and still find it necessary to include the notion of an SDLC (albeit for a different section and not relating to this particular section). A lot of the literature around security testing references the SDLC and it is an important concept towards understanding the application of security testing.

I'm hoping to keep some kind of reference to allow a reader to learn more about SDLCs, just wondering whether you think it would be best done in an info box or as a footnote/reference?

Reply:
I don't think any reader of this article would have a general idea of what SDLC is? Anyway, you can just hyperlink the word SDLC to the relevant article.